TotalCrossVM/src/launchers/linux/Launcher.c

```diff
@@ -58,19 +58,23 @@ int main(int argc, const char *argv[])
 {
    char cmdline[512];
    xmemzero(cmdline,sizeof(cmdline));
+   int argvIndex = 0;
    if (argv)
    {
-      xstrcpy(cmdline, argv[0]);
+      if (argc > 1 && strEq(argv[0] + xstrlen(argv[0]) - 8, "Launcher")) {
```

Comment:
I think this may overflow. I suggest the use of `strncmp` which could use the `strlen` as `size` argument to avoid this risk.